Build failed because two lint jobs ran simultaneously

**Describe the bug**

This requires more triaging since the builds should not be sharing any directories but [build 1355](https://buildkite.com/opstrace/scheduled-main-builds/builds/1355#ed9dcab4-3c5a-4c15-bece-57df9ba24626/1532-1630) failed with:

```
[2020-12-12T20:10:08Z] (cd go/ && golangci-lint run)
[2020-12-12T20:10:13Z] ERRO Parallel golangci-lint is running
```

[Build 3271](https://buildkite.com/opstrace/prs/builds/3271#3fb10326-e962-4c2b-92ec-29d4fb13ace4/1594-1689) was running the lint job at the same time:

```
[2020-12-12T20:10:07Z] (cd go/ && golangci-lint run)
[2020-12-12T20:10:16Z] make[1]: Leaving directory '/build'
```

I believe we're still seeing this issue: https://buildkite.com/opstrace/prs/builds/4376#57bdfe0a-97ad-4a59-ac96-7eec8bda6623
```

[2021-05-28T14:28:57Z]  \|   297:5  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
--
&nbsp; | [2021-05-28T14:28:57Z]  \|   306:5  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
&nbsp; | [2021-05-28T14:28:57Z]  \|   315:5  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
&nbsp; | [2021-05-28T14:28:57Z]  \|
&nbsp; | [2021-05-28T14:28:57Z]  \| /build/lib/kubernetes/src/reconciliation/reconcile.ts
&nbsp; | [2021-05-28T14:28:57Z]  \|   215:11  warning  '_pvc' is defined but never used  @typescript-eslint/no-unused-vars
&nbsp; | [2021-05-28T14:28:57Z]  \|
&nbsp; | [2021-05-28T14:28:57Z]  \| ✖ 27 problems (0 errors, 27 warnings)
&nbsp; | [2021-05-28T14:28:57Z]  \|
&nbsp; | Done in 73.55s.
&nbsp; | [2021-05-28T14:28:57Z] # Also see https://github.com/opstrace/opstrace/issues/166
&nbsp; | [2021-05-28T14:28:57Z] (cd go/ &amp;&amp; golangci-lint run --allow-parallel-runners)
&nbsp; | [2021-05-28T14:29:57Z] ERRO Running error: context loading failed: failed to load packages: timed out to load packages: context deadline exceeded
&nbsp; | [2021-05-28T14:29:57Z] ERRO Timeout exceeded: try increasing it by passing --timeout option
&nbsp; | [2021-05-28T14:29:58Z] make[1]: *** [Makefile:236: lint-codebase.go] Error 4
&nbsp; | [2021-05-28T14:29:58Z] make[1]: Leaving directory '/build'
&nbsp; | [2021-05-28T14:29:58Z] make: *** [Makefile:740: preamble] Error 2
&nbsp; | [2021-05-28T14:29:59Z] make: *** [ci-preamble] Error 2
&nbsp; | [2021-05-28T14:29:59Z] 🚨 Error: The command exited with status 2


```

--- a/Makefile
+++ b/Makefile
@@ -233,7 +233,7 @@ lint-codebase.js:
 
 lint-codebase.go:
 	# Also see https://github.com/opstrace/opstrace/issues/166
-	(cd go/ && golangci-lint run --allow-parallel-runners)
+	(cd go/ && golangci-lint run --timeout 3m --allow-parallel-runners)
 
 .PHONY: cli-tsc
 cli-tsc: set-build-info-constants

--- a/containers/ci/opstrace-ci.Dockerfile
+++ b/containers/ci/opstrace-ci.Dockerfile
@@ -89,7 +89,7 @@ ENV PATH /usr/local/go/bin:$GOPATH/bin:$PATH
 
 # Set up golanglint-ci in the container image.
 RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
-    sh -s -- -b /usr/local/bin v1.38.0
+    sh -s -- -b /usr/local/bin v1.40.1
 
 # Set up markdownlint in the container image so that we can lint right away! :)
 RUN npm install -g markdownlint-cli@0.26.0

